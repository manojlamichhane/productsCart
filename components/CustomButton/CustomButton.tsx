"use client";
import { CustomButtonProps } from "@/types";
import React from "react";

const CustomButton = ({
  title,
  containerStyles,
  handleClick,
  btnType,
}: CustomButtonProps) => {
  return (
    <button
      disabled={false}
      type={btnType}
      data-testid="button"
      onClick={handleClick}
      className={`custom-btn ${containerStyles}`}
    >
      <span>{title}</span>
    </button>
  );
};

export default CustomButton;
