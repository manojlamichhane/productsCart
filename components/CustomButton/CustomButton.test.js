import React from "react";
import "@testing-library/jest-dom";
import { render, screen } from "@testing-library/react";
import CustomButton from "./CustomButton";

describe("Button test", () => {
  test("Render check", () => {
    const onClick = jest.fn();
    const { asFragment } = render(
      <CustomButton title="Hello" handleClick={onClick} />
    );
    expect(screen.getByTestId("button")).toBeInTheDocument();
    expect(screen.getByRole("button")).toBeInTheDocument();
    expect(asFragment()).toMatchSnapshot();
  });
  test("Should display a button with given name", () => {
    const onClick = jest.fn();
    render(<CustomButton title="Hello" handleClick={onClick} />);

    expect(screen.getByText("Hello")).toBeInTheDocument();
  });
});
