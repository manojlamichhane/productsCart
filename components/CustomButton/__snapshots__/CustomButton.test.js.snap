// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button test Render check 1`] = `
<DocumentFragment>
  <button
    class="custom-btn undefined"
    data-testid="button"
  >
    <span>
      Hello
    </span>
  </button>
</DocumentFragment>
`;
